src/k8s/request.js

```diff
@@ -366,26 +366,31 @@ const addNetworks = (vm, template, getSetting, networks) => {
 };
 
 const addCloudInit = (vm, defaultDisk, getSetting) => {
-  if (getSetting(USE_CLOUD_INIT_KEY)) {
-    const existingCloudInitVolume = getCloudInitVolume(vm);
-    const cloudInit = new CloudInit({
-      volume: existingCloudInitVolume,
+  if (!getSetting(USE_CLOUD_INIT_KEY)) {
+    vm.spec.template.spec.hostname = getSetting(NAME_KEY);
+    return;
+  }
+
+  const existingCloudInitVolume = getCloudInitVolume(vm);
+  const cloudInit = new CloudInit({
+    volume: existingCloudInitVolume,
+  });
+
+  if (getSetting(USE_CLOUD_INIT_CUSTOM_SCRIPT_KEY)) {
+    cloudInit.setUserData(getSetting(CLOUD_INIT_CUSTOM_SCRIPT_KEY));
+  } else {
+    cloudInit.setPredefinedUserData({
+      hostname: getSetting(HOST_NAME_KEY),
+      sshAuthorizedKeys: getSetting(AUTHKEYS_KEY),
     });
+  }
 
-    if (getSetting(USE_CLOUD_INIT_CUSTOM_SCRIPT_KEY)) {
-      cloudInit.setUserData(getSetting(CLOUD_INIT_CUSTOM_SCRIPT_KEY));
-    } else {
-      cloudInit.setPredefinedUserData({
-        hostname: getSetting(HOST_NAME_KEY),
-        sshAuthorizedKeys: getSetting(AUTHKEYS_KEY),
-      });
-    }
+  vm.spec.template.spec.hostname = getSetting(HOST_NAME_KEY) || getSetting(NAME_KEY);
```

Comment:
you can move this line to the beginning and remove the one at line 370

Reply:
Done!